back-end/app/Http/Controllers/IngredientsController.php

```diff
@@ -1,30 +1,68 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\Ingredients;
+use App\Models\Ingredient;
+use Illuminate\Http\JsonResponse;
+
 class IngredientsController extends Controller
 {
     /**
-     * Display a listing of the resource.
+     * Get all ingredients from DB.
      *
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\JsonResponse
      */
-    public function index()
+    public function index(): JsonResponse
     {
-        return response()->json(['code' =>"Tutti gli ingredienti"],200);
+        $data = Ingredient::getAllIngredient();
+        return response()->json(['data' => $data ],self::OK);
 
     }
 
     /**
-     * Display the specified resource.
+     * Get the specified ingredients from DB.
      *
      * @param  int  $id
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function show($id): JsonResponse
+    {
+        $data = Ingredient::getSpecifiedIngredients($id);
+        return response()->json(['data' => $data ],self::OK);
+
+    }
+
+    /**
+     * Method to create an Ingredient.
+     *
+     * @param Ingredients $request
+     *
+     * @return JsonResponse
      */
-    public function show($id)
+    public function create(Ingredients $request): JsonResponse
     {
-        return response()->json(['code'=>"Restituzione ingrediente"],200);
+        $name = $request->input('name');
+        $newIngredient = Ingredient::createIngredients($name);
+        return response()->json(['data' => $newIngredient],self::OK);
```

Comment:
Stessa cosa qua e immagino anche sotto. Impostati PHPCS fixer nel caso :)

Reply:
mi pare di averli aggiunti. Si il fixer lo metto dammi tempo. :D No gli unit test li mettiamo in qua. Voglio controllare come inserire un DB di test.

Reply:
Ok, allora metti sul titolo WIP, ho aggiunto un tool che se lo scrivi rende questa pull request not mergeable.